pkg/router/template/util/rewritetarget/rewritetarget_test.go

```diff
@@ -0,0 +1,94 @@
+package rewritetarget_test
+
+import (
+	"github.com/openshift/router/pkg/router/template/util/rewritetarget"
```

Comment:
It's not the same package. The tests are in their own package. i.e., `rewritetarget_test`.

Reply:
Ah, right, I was looking at the directories. So, what's the idea behind? To test without the access to any private variables? Like if it was the user of the package?

I see this pattern is not quite uniformly implemented in `util` package (e.g. [haproxy](https://github.com/openshift/router/blob/4933f71131a3f4fc9f4ca0d93449569e315b74b2/pkg/router/template/util/haproxy/map_entry_test.go#L1)).

Reply:
>     Encapsulation: It guarantees tests can only access exported package members, mirroring how external packages interact with rewritetarget. This approach reinforces the integrity of the package's encapsulation.
> 
>     API Design: Black-box testing inherently leads to clearer and more stable APIs by focusing on the package's usability rather than its internals.
> 
>     Resilience: Tests are less prone to break due to internal changes, as they're tied to the public API. This helps to reduce maintenance.
> 
>     User Perspective: It provides insights into how the package is used, highlighting potential areas for improvement from the end-user’s standpoint.

I see. Yes, I think it makes sense for `SanitizeInput` function.

Reply:
Right, Andy and I had this discussion in slack. It's an interesting perspective we should consider when writing unit tests.